### PR TITLE
Added new top level resource for verifying VNET for ASE

### PR DESCRIPTION
**Pull request by hforeste**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json [hunk 1]
```diff
@@ -307,6 +307,51 @@
...
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Web/verifyHostingEnvironmentVnet": {
+      "post": {
+        "summary": "Verifies if this VNET is compatible with an App Service Environment by analyzing the Network Security Group rules",
+        "description": "Verifies if this VNET is compatible with an App Service Environment by analyzing the Network Security Group rules",
+        "operationId": "VerifyHostingEnvironmentVnet",
+        "consumes": [
+          "application/json",
+          "text/json",
```
Comment (sergey-shandar) [inline specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json]: Does your service support all of these types? AFAIK, ARM recommends only "allpication/json".

File: specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json [hunk 2]
```diff
@@ -555,6 +600,10 @@
         }
       }
     },
+    "Object": {
```
Comment (sergey-shandar) [inline specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json]: Is there a reason to create an alias for an object type?

File: specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json [hunk 3]
```diff
@@ -307,6 +307,51 @@
...
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Object"
```
Comment (sergey-shandar) [inline specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json]: Is it possible to provide more information about what is the response structure? For example, does it have some common properties?

File: specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json [hunk 4]
```diff
@@ -967,6 +1016,35 @@
           "type": "string"
         }
       }
+    },
+    "VnetParameters": {
+      "description": "",
```
Comment (sergey-shandar) [inline specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json]: Could you provide a 'description' for the type? The description is used to generate API comments and documentation so it is useful for our customers.

**Review by sergey-shandar — CHANGES_REQUESTED** [review]

File: specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json [hunk 5]
```diff
@@ -307,6 +307,51 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Web/verifyHostingEnvironmentVnet": {
+      "post": {
+        "summary": "Verifies if this VNET is compatible with an App Service Environment by analyzing the Network Security Group rules",
```
Comment (hforeste) [inline specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json]: I shortened the summary in the next iteration.
Our tooling auto generates this swagger file hence why the summary and description are the same. Similar to the other APIs in the file.

File: specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json [hunk 6]
```diff
@@ -307,6 +307,51 @@
...
+          "application/xml",
+          "text/xml",
+          "application/x-www-form-urlencoded"
+        ],
+        "produces": [
+          "application/json",
+          "text/json",
+          "application/xml",
+          "text/xml"
+        ],
```
Comment (sergey-shandar) [inline specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json]: @hforeste still the same question, does your service support all of the content types (text/json, application/xml)? If supported content type are the same, could you place them to the top level of the swagger?
@ravbhatnagar are you ok with these content types?

File: specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json [hunk 7]
```diff
@@ -967,6 +1012,84 @@
...
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "VnetValidationFailureDetails resource specific properties",
+          "properties": {
+            "failed": {
```
Comment (sergey-shandar) [inline specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json]: Could you add description to the property?

File: specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json [hunk 8]
```diff
@@ -967,6 +1012,84 @@
...
+              "type": "boolean"
+            },
+            "failedTests": {
+              "type": "array",
```
Comment (sergey-shandar) [inline specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json]: same here: description

File: specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json [hunk 9]
```diff
@@ -967,6 +1012,84 @@
...
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "VnetValidationTestFailure resource specific properties",
+          "properties": {
+            "testName": {
+              "type": "string"
+            },
+            "details": {
```
Comment (sergey-shandar) [inline specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json]: same here, 'description'

**Review by sergey-shandar — CHANGES_REQUESTED** [review]
Could you provide x-ms-example(s) for the new operation?

File: specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json (hunk 6 above)
Comment (hforeste) [inline specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json]: @sergey-shandar , our tooling automatically adds these content types. We're planning to fix this here in the near future.

Comment (sergey-shandar) [inline specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json]: @hforeste Excellent! Do you have a ticket for this issue so we can track it?

Comment (hforeste) [inline specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json]: I'll have to check with @naveedaz, not sure where this work item might be tracked. Navy, do you have a work item tracked for this?

**Comment by hforeste** [comment]
**_Could you provide x-ms-example(s) for the new operation?_**
->I think we can address this afterwards.

**Comment by sergey-shandar** [comment]
@hforeste do you need help with examples?

File: specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json (hunk 6 above)
Comment (ravbhatnagar) [reply specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json]: @hforeste - ARM does not support content types other than application/json. So, please remove them. Even though these are generated by a tool, will it be a problem if you manually fix these?

File: specification/web/resource-manager/Microsoft.Web/2016-03-01/examples/VerifyHostingEnvironmentVnet.json [hunk 10]
```diff
@@ -0,0 +1,24 @@
+{
+  "parameters": {
+      "subscriptionId": "34adfa4f-cedf-4dc0-ba29-b6d1a69ab345",
+      "api-version": "2016-03-01",
+      "vNetParameters": {
+        "properties": { 
+            "vnetResourceGroup": 'vNet123rg',
```
Comment (sergey-shandar) [inline specification/web/resource-manager/Microsoft.Web/2016-03-01/examples/VerifyHostingEnvironmentVnet.json]: JSON errors. Please, use double quotes "".

File: specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json (hunk 6 above)
Comment (hforeste) [reply specification/web/resource-manager/Microsoft.Web/2016-03-01/ResourceProvider.json]: No, it should not be a problem. Thanks!

**Review by sergey-shandar — CHANGES_REQUESTED** [review]

File: specification/web/resource-manager/Microsoft.Web/2016-03-01/examples/VerifyHostingEnvironmentVnet.json [hunk 11]
```diff
@@ -0,0 +1,24 @@
+{
+  "parameters": {
+      "subscriptionId": "34adfa4f-cedf-4dc0-ba29-b6d1a69ab345",
+      "api-version": "2016-03-01",
+      "vNetParameters": {
```
Comment (sergey-shandar) [inline specification/web/resource-manager/Microsoft.Web/2016-03-01/examples/VerifyHostingEnvironmentVnet.json]: the property should be called "parameters".

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**